components/wpa_supplicant/src/common/sae.c

```diff
@@ -594,9 +594,7 @@ static struct crypto_ec_point * sswu(struct crypto_ec *ec, int group,
 
 	prime = crypto_ec_get_prime(ec);
 	prime_len = crypto_ec_prime_len(ec);
-	/* Value of 'a' defined for curve secp256r1 in 'y^2 = x^3 + ax + b' */
-	uint8_t buf[32] = {0xff,0xff,0xff,0xff,0x00,0x00,0x00,0x01,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xff,0xfc};
-	a = crypto_bignum_init_set(buf, 32);
+	a = crypto_ec_get_a_allocated(ec);
```

Comment:
```suggestion
	a = crypto_ec_get_b(ec);
```

let's keep the function name similar to upstream https://w1.fi/cgit/hostap/tree/src/crypto/crypto.h#n787

Reply:
OK. Care should be taken when using the API: don't forget to free it.

Reply:
actually that code should be added in `crypto_ec_deinit()`

Reply:
> actually that code should be added in `crypto_ec_deinit()`

You mean embed the A in struct crypto_ec?

Reply:
On second thought, we can do this as well
```
const struct crypto_bignum * crypto_ec_get_a(struct crypto_ec *e)
{
    mbedtls_ecp_group *grp = (mbedtls_ecp_group *)e; 
#ifdef MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED
    if (mbedtls_ecp_get_type(grp) != MBEDTLS_ECP_TYPE_SHORT_WEIERSTRASS) {
        return NULL;
    }    
    /* Handle Mbed TLS quirk.
     *
     * Mbed TLS default ECP implementation is using grp->A = NULL to represent A = -3 for
     * Short Weierstrass curves(e.g. P-256) thus accessing A needs some tweaking.
     *
     * See mbedtls/ecp.h for details. */
     if (mbedtls_ecp_group_a_is_minus_3(grp)) {
         mbedtls_mpi_sub_int(&grp->A, &grp->P, 3);
     }    
#endif
    return (const struct crypto_bignum *) (&grp->A);
}
```
and keep `sswu()` implementation same as upstream.

Reply:
This may not be a good solution. Doing so would interfere with the internal logic of MbedTLS, and the MbedTLS maintainers also do not approve of this approach. See https://github.com/Mbed-TLS/mbedtls/pull/6999#discussion_r1286697700.
Maybe we could embed parameter A in struct crypto_ec like this:
```c
struct crypto_ec {
    /* group must be the first member */
    mbedtls_ecp_group group;
    /* valid only if group is of Short Weierstrass type */
    mbedtls_mpi group_parameter_a;
};
```
And load A in crypto_ec_init(), while free it in crypto_ec_deinit(). In this way, sswu() can be kept same as upstream
and crypto_ec_get_a() can behave the same as upstream. This seems more robust.  However, this will introduce some unnecessary runtime memory overhead when not doing H2E operation.

Reply:
Sorry, I missed this. Currently changes looks okay IMO, will raise internal MR. I don't want to embed A in ec struct since that will unnecessary increase size.

Reply:
OK